auth/grants.go

```diff
@@ -48,6 +51,116 @@ func (c *RoomConfiguration) UnmarshalJSON(data []byte) error {
 	return protojson.Unmarshal(data, (*livekit.RoomConfiguration)(c))
 }
 
+// CheckCredentials checks if the room configuration contains sensitive credentials
+// and returns an error if it does.
+//
+// This is used to prevent sensitive credentials from being leaked to the client.
+// It is not used to validate the credentials themselves, as that is done by the
+// egress service.
+func (c *RoomConfiguration) CheckCredentials() error {
+	if c.Egress == nil {
+		return nil
+	}
+
+	if c.Egress.Participant != nil {
+		for _, output := range c.Egress.Participant.FileOutputs {
+			if err := checkOutputForCredentials(output.Output); err != nil {
+				return err
```

Comment:
https://pkg.go.dev/slices#ContainsFunc

Cleaner to use this instead of the for loop

Reply:
Ah yes good point, let me try to come up with something

Reply:
https://github.com/livekit/protocol/commit/03e2559a4b7ec09c706dc2bafc442f51e31a8afb

How does this look? All tests pass.

Reply:
personally, I'm not a huge fan of using reflections when alternatives exists. it's less code, but much harder to read.